crates/net/rpc/src/net/mod.rs

```diff
@@ -0,0 +1,35 @@
+use jsonrpsee::core::RpcResult as Result;
+use reth_network::NetworkHandle;
+use reth_rpc_api::NetApiServer;
+use reth_rpc_types::PeerCount;
+
+/// `Net` API implementation.
+///
+/// This type provides the functionality for handling `net_` related requests.
+pub struct NetApi {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+    /// The devp2p network ID
+    network_id: String,
```

Comment:
I suggest we replace this with `EthApi` and use `EthApi::chain_id()` for `fn version`

Reply:
dyn EthApi trait of the impl? the latter would require its generics 😓

Reply:
Ah right,

okay, so what we should do here is, introduce another `EthApi` trait, but which is a subset of `EthApiServer` trait but without generics, essentially everything that is already in `impl EthApi {`

Reply:
Or we could use `Box< dyn EthApiServer>` which is perhaps easier, but I guess we still want a separate `EthApi` trait

Reply:
EthApi trait IMO